crates/oxc_linter/src/config/config_builder.rs

```diff
@@ -326,6 +326,7 @@ impl ConfigStoreBuilder {
         let new_rules = self
             .rules
             .iter()
+            .sorted_by_key(|x| format!("{}/{}", x.plugin_name(), x.name()))
```

Comment:
I don't know how `x.plugin_name` works for rules that don't have a plugin, such as `for-direction`. Testing shows that it works without an error, but I'm not sure if there should be some sort of check here for that scenario.

Reply:
`format` allocates. You can use a tuple instead: https://stackoverflow.com/a/70194856